plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1830,7 +1830,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
    }
 
    // create speculative blocks at regular intervals, so we create blocks with "current" block time
-   _pending_block_deadline = now + fc::microseconds(config::block_interval_us);
+   _pending_block_deadline = block_timing_util::calculate_producing_block_deadline(_produce_block_cpu_effort, block_time);
```

Comment:
I believe that `calculate_producing_block_deadline` will return `std::nullopt` if `producers.empty()`.
Is it OK to not have a deadline in that case?

Reply:
This is calling `calculate_producing_block_deadline` not `calculate_producer_wake_up_time`

Reply:
Oh, my bad, thanks!